Ctl 006 update contact screen

heupotato opened the pull request:
@lngyeen Pull Request: Update-contact Screen

heupotato committed:
Update type of gender due to change of input value

Change type of 'gender' variable from 'int' to 'String'

heupotato committed:
Update action when click 'edit' in 'contact list screen'

heupotato committed:
Update Contact Screen

Update function in mock_data to get contact by ID
Create update contact screen

heupotato committed:
Update workspace.xml

heupotato committed:
Apply changes based on gender's type change

lngyeen commented on lib/ui/screens/update_contact_screen.dart:
```diff
@@ -0,0 +1,166 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_contact_list/mockdata/mock_contact.dart';
+
+class UpdateContact extends StatefulWidget {
```
Change class name to UpdateContactScreen

lngyeen replied:
Change class name to UpdateContactScreen

lngyeen commented on lib/ui/screens/update_contact_screen.dart:
```diff
@@ -0,0 +1,166 @@
+        return TextFormField(
+            decoration: InputDecoration(labelText: "Last name"),
+            initialValue: oldLastName,
+            validator: (String ? value) {
```
Use validator from Validator class

lngyeen commented on lib/ui/screens/update_contact_screen.dart:
```diff
@@ -0,0 +1,166 @@
+        return TextFormField(
+            decoration: InputDecoration(labelText: "Phone Number"),
+            initialValue: oldPhone,
+            validator: (String ? value) {
```
Use validator from Validator class

lngyeen commented on lib/ui/screens/update_contact_screen.dart:
```diff
@@ -0,0 +1,166 @@
+            print("Nahhhh"); 
+        }
+        else{
+            formKey.currentState!.save();
```
Should not use force unwrap optional value (!).
Change to formKey.currentState?.save();

lngyeen commented on lib/ui/screens/update_contact_screen.dart:
```diff
@@ -0,0 +1,166 @@
+    @override
+    Widget build(BuildContext context) {
+        //get data of old contact by its id 
+        final oldContact = MockContact.mockContact(widget.id); 
```
Change to new function `MockContact.mock(widget.id); `

heupotato committed:
Use new mock data method

heupotato committed:
Update "update_contact" screen

Consist of:
- Change name of class, from UpdateContact to UpdateContactScreen
- Change List<String> Gender to enum
- Change type of gender field in class UpdateContactScreenState
- Use Validator class

heupotato committed:
Update workspace.xml
